libraries/standard/utilities/include/iot_logging_setup.h

```diff
@@ -0,0 +1,272 @@
+ */
+
+/**
+ * @def IotLogDebugWithArgs( pFormat, ...  )
+ * @brief Abbreviated logging macro for messages with arguments at level #IOT_LOG_DEBUG.
+ *
+ * Equivalent to:
+ * @code{c}
+ * IotLog( IOT_LOG_DEBUG, pFormat, ... )
+ * @endcode
+ */
+
+/* Check that LIBRARY_LOG_LEVEL is defined and has a valid value. */
+#if !defined( LIBRARY_LOG_LEVEL ) ||           \
+    ( ( LIBRARY_LOG_LEVEL != IOT_LOG_NONE ) && \
+    ( LIBRARY_LOG_LEVEL != IOT_LOG_ERROR ) &&  \
+    ( LIBRARY_LOG_LEVEL != IOT_LOG_WARN ) &&   \
+    ( LIBRARY_LOG_LEVEL != IOT_LOG_INFO ) &&   \
+    ( LIBRARY_LOG_LEVEL != IOT_LOG_DEBUG ) )
+    #error "Please define LIBRARY_LOG_LEVEL as either IOT_LOG_NONE, IOT_LOG_ERROR, IOT_LOG_WARN, IOT_LOG_INFO, or IOT_LOG_DEBUG."
+/* Check that LIBRARY_LOG_NAME is defined and has a valid value. */
+#elif !defined( LIBRARY_LOG_NAME )
+    #error "Please define LIBRARY_LOG_NAME."
+#else
+    #if LIBRARY_LOG_LEVEL != IOT_LOG_NONE
+        #if !defined( IotLog )
+            #error "Please define the common logging interface macro, IotLog(messageLevel, pFormat, ...)."
+        #endif
+    #endif
+
+    #if LIBRARY_LOG_LEVEL == IOT_LOG_DEBUG
+        /* All log levels will logged. */
+        #define IotLogError( pFormat )                 IotLog( IOT_LOG_ERROR, "%s", pFormat )
+        #define IotLogErrorWithArgs( pFormat, ... )    IotLog( IOT_LOG_ERROR, pFormat, __VA_ARGS__ )
+        #define IotLogWarn( pFormat )                  IotLog( IOT_LOG_WARN, "%s", pFormat )
+        #define IotLogWarnWithArgs( pFormat, ... )     IotLog( IOT_LOG_WARN, pFormat, __VA_ARGS__ )
+        #define IotLogInfo( pFormat )                  IotLog( IOT_LOG_INFO, "%s", pFormat )
+        #define IotLogInfoWithArgs( pFormat, ... )     IotLog( IOT_LOG_INFO, pFormat, __VA_ARGS__ )
+        #define IotLogDebug( pFormat )                 IotLog( IOT_LOG_DEBUG, "%s", pFormat )
+        #define IotLogDebugWithArgs( pFormat, ... )    IotLog( IOT_LOG_DEBUG, pFormat, __VA_ARGS__ )
+
+/* Remove references to IotLog from the source code if logging is disabled. */
+    #elif LIBRARY_LOG_LEVEL == IOT_LOG_INFO
+        /* Debug messages will not be logged. All other macros will call IotLog */
+        #define IotLogError( pFormat )                 IotLog( IOT_LOG_ERROR, "%s", pFormat )
+        #define IotLogErrorWithArgs( pFormat, ... )    IotLog( IOT_LOG_ERROR, pFormat, __VA_ARGS__ )
+        #define IotLogWarn( pFormat )                  IotLog( IOT_LOG_WARN, "%s", pFormat )
+        #define IotLogWarnWithArgs( pFormat, ... )     IotLog( IOT_LOG_WARN, pFormat, __VA_ARGS__ )
+        #define IotLogInfo( pFormat )                  IotLog( IOT_LOG_INFO, "%s", pFormat )
+        #define IotLogInfoWithArgs( pFormat, ... )     IotLog( IOT_LOG_INFO, pFormat, __VA_ARGS__ )
+        #define IotLogDebug( pFormat )
+        #define IotLogDebugWithArgs( pFormat, ... )
+
+    #elif LIBRARY_LOG_LEVEL == IOT_LOG_WARN
+        /* Only "Warning" and IOT_LOG_ERROR messages will be logged.*/
```

Comment:
Should EITHER be "Only Warning and Error messages ..." OR "Only IOT_LOG_WARN and IOT_LOG_ERROR messages ..." - My preference being the former to keep it consistent with above comments.

Reply:
Fixed inconsistency issues